### PR TITLE
Remove hardcoded certificates from Kubernetes installation files

### PR DESCRIPTION
**Pull request by scholzj**
Strimzi currently offers 3 different sets of installation files:
1) For clusters using Cert-Manager
2) For OpenShift clusters
3) For plain Kubernetes without any addons

The first two installation methods do not contain any hardcoded certificate and use either Cert-Manager or OpenShift-only features to have the certificates generated on the fly. But the third method has currently hardcoded certificates. This is not entirely nice. While in this case it might not be seen as a major security issue, it often triggers different scanning bots because of the secret certificates etc. 

This PR replaces the hardcoded certificates and removes the scripts for generating the certificates with `cfssl`. It replaces them with guide based on OpenSSL which leads users through generating the certificates and updating the YAML files. The OpenSSL commands are relatively simple. And why I'm personally a big fan of `cfssl`, I assume it is much more niche tool compared to OpenSSL.

### CODE REVIEW

**Review by Frawless — APPROVED** [review]

File: packaging/install/kubernetes/README.md [hunk 1]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
 So the web-hook service needs to have HTTPS support.
 And the CA of the certificate used for this service needs to be specified in the `ValidatingWebhookConfiguration` resource.
 
-This directory contains sample files with pre-generated certificates.
-As long as you don't change the namespace name or any service / pod names, you can just install them.
+This directory contains sample installation files without the certificates.
+You can generate the certificates by following the guide below and update the installation files with the,
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:9]:
```suggestion
The following procedures describe how to generate certificates and add them to installation files.
```

File: packaging/install/kubernetes/README.md [hunk 2]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
 
-Additionally, in the `webhook-certificates` subdirectory, you have files which you can use to generate your own certificates using the [`cfssl` tool](https://github.com/cloudflare/cfssl).
-In case you decide to generate your own certificates, you can use the script to generate them and then you have to update the `040-Secret.yaml` and `070-ValidatingWebhookConfiguration.yaml`.
-Remember, that both resources contain the certificates encoded in base64.
\ No newline at end of file
+## Generating the certificate using OpenSSL
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:11]:
```suggestion
## Generating certificates using OpenSSL
```

File: packaging/install/kubernetes/README.md [hunk 3]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+
+You can use the OpenSSL utility to generate the TLS certificate for the Strimzi Drain Cleaner webhook.
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:13]:
```suggestion
Use the OpenSSL TLS management tool to generate the TLS certificate for the Strimzi Drain Cleaner webhook.
```

File: packaging/install/kubernetes/README.md [hunk 4]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+The steps below have been tested with OpenSSL 1.1.1 and should work on Linux, MacOS or in the Windows Subsystem for Linux.
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:14]:
```suggestion
The steps below have been tested with OpenSSL 1.1.1 and should work on Linux, MacOS, or in the Windows Subsystem for Linux.
```

File: packaging/install/kubernetes/README.md [hunk 5]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+
+1) Create a subdirectory `tls-certificate` where we will generate the certificate files and enter it:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:16]:
```suggestion
1) Create and navigate to a subdirectory called `tls-certificate`:
```

File: packaging/install/kubernetes/README.md [hunk 6]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+   mkdir tls-certificate
+   cd tls-certificate
+   ```
+2) Generate the CA public and private keys:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:21]:
```suggestion
2) Generate a CA public certificate and private key in the  `tls-certificate` directory:
```

File: packaging/install/kubernetes/README.md [hunk 7]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+   openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=Strimzi Drain Cleaner CA"
+   ```
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:24]:
```suggestion 
    ```
    openssl req -nodes -new -x509 -keyout ca.key -out ca.crt -subj "/CN=Strimzi Drain Cleaner CA"
    ```
    
   A `ca.crt` and `ca.key` file is created. 
```

File: packaging/install/kubernetes/README.md [hunk 8]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+3) Generate the private key for the Strimzi Drain Cleaner:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:25]:
```suggestion
3) Generate the private TLS key for the Strimzi Drain Cleaner:
```

File: packaging/install/kubernetes/README.md [hunk 9]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+   openssl genrsa -out tls.key 2048
+   ```
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:28]:
```suggestion
    ```
    openssl genrsa -out tls.key 2048
    ```
    
    A `tls.key` file is created.
```

Comment (PaulRMellor) [reply packaging/install/kubernetes/README.md:28]: If you make the changes, check the formatting on the last couple of suggestions. They weren't working as expected in the comments boxes

File: packaging/install/kubernetes/README.md [hunk 10]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+4) Generate the Certificate Signing Request and sign it with the CA from step 1:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:29]:
```suggestion
4) Generate a Certificate Signing Request and sign it by adding the CA public certificate (`ca.crt`) you generated:
```

File: packaging/install/kubernetes/README.md [hunk 11]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+   openssl req -new -key tls.key -subj "/CN=strimzi-drain-cleaner.strimzi-drain-cleaner.svc" \
+        | openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -extfile <(printf "subjectAltName=DNS:strimzi-drain-cleaner.strimzi-drain-cleaner.svc") -out tls.crt
+   ```
+   If you plan to change the name of the Strimzi Drain Cleaner service or install it into a different namespace, you have to change the common name (CN) of the certificate.
+   The CN should follow the pattern `<SERVICE_NAME>.<NAMESPACE_NAME>.svc`.
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:35]:
```suggestion
   The CN must follow the pattern `<SERVICE_NAME>.<NAMESPACE_NAME>.svc`.
```

File: packaging/install/kubernetes/README.md [hunk 12]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+5) The `tls-certificate` directory should now contain several certificate files which we will use in the installation files.
+   You can exit the `tls-certificate` directory now.
+   ```
+   cd ..
+   ```
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:40]: I'm not sure we need step 5?

File: packaging/install/kubernetes/README.md [hunk 13]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+
+## Updating the installation files with the generated certificates
+
+This section expects that you have the following certificate files generated in the previous section.
+If you generated your certificates in a different way or on a different path, you should update the path in the commands.
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:45]:
```suggestion
After you have generated the certificates you need, update the installation files.
This procedure assumes that you used the previous procedure to generate the certificate files.
If you generated your certificates in a different way or on a different path, you should update the path in the commands.
```

File: packaging/install/kubernetes/README.md [hunk 14]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+
+1) Edit the [`070-ValidatingWebhookConfiguration.yaml`](070-ValidatingWebhookConfiguration.yaml) file and update the `caBundle` field to contain the Base64 encoded public key of your CA.
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:47]:
```suggestion
1) Edit the `caBundle` field in the [`070-ValidatingWebhookConfiguration.yaml`](070-ValidatingWebhookConfiguration.yaml) installation file to specify a Base64 encoded public key of your CA.
```

File: packaging/install/kubernetes/README.md [hunk 15]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   You can get the Base64 encoded public key for example using the `base64` utility:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:48]:
```suggestion
   You can use the `base64` utility to get the Base64 encoded public key:
```

File: packaging/install/kubernetes/README.md [hunk 16]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+   base64 tls-certificate/ca.crt
+   ```
+   Use the output of one of these commands as the `caBundle`.
+   You YAML should look similar to this:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:53]:
```suggestion
   After adding the Base64 encoded public key, your YAML should look similar to this:
```

File: packaging/install/kubernetes/README.md [hunk 17]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+   ```
+3) Generate the private key for the Strimzi Drain Cleaner:
+   ```
+   openssl genrsa -out tls.key 2048
+   ```
+4) Generate the Certificate Signing Request and sign it with the CA from step 1:
+   ```
+   openssl req -new -key tls.key -subj "/CN=strimzi-drain-cleaner.strimzi-drain-cleaner.svc" \
+        | openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -extfile <(printf "subjectAltName=DNS:strimzi-drain-cleaner.strimzi-drain-cleaner.svc") -out tls.crt
+   ```
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:33]:
```suggestion
    ```
    A `tls.crt` file is created.
```

File: packaging/install/kubernetes/README.md [hunk 18]
```diff
@@ -5,9 +5,97 @@ Kubernetes requires that `ValidationWebhooks` are secured using TLS.
...
+3) Create the Secret named `strimzi-drain-cleaner` with the `tls.crt` and `tls.key` files.
+   This Secret will be used by the Strimzi Drain Cleaner deployment.
+   You can create the Secret using `kubectl`:
+   ```
+   kubectl create secret tls strimzi-drain-cleaner \
+       -n strimzi-drain-cleaner  \
+       --cert=tls-certificate/tls.crt \
+       --key=tls-certificate/tls.key
+   ```
+   The resulting Secret should look similar to this:
```
Comment (PaulRMellor) [inline packaging/install/kubernetes/README.md:78]:
```suggestion
3) Use `kubectl` to create a secret named `strimzi-drain-cleaner` with the `tls.crt` and `tls.key` files you generated:
   
     ```
   kubectl create secret tls strimzi-drain-cleaner \
       -n strimzi-drain-cleaner  \
       --cert=tls-certificate/tls.crt \
       --key=tls-certificate/tls.key
    ```
   This secret is used by the Strimzi Drain Cleaner deployment.
   The resulting Secret should look similar to this:
```

**Review by PaulRMellor — APPROVED** [review]
Looks good. Some suggestions
If applied, check the markdown placement on the suggestions with code/file fragments as they were not showing correctly in the comment

**Pull request merged**